### DISCUSSION TITLE
Support of Visa Payment Transactions?

### DISCUSSION THREAD

**Discussion by V0xxel** [pytr-org/pytr#94] [Q&A]
I saw that the issue #53 about adding visa transactions was closed, but i cant find any documents for the **payment** transactions made with the visa card. 
_(Saveback order resulting from these payments however **are** downloaded into a dedicated "Saveback" folder.)_

I also tried redownloading all documents to a blank folder, since the new version introduced some changes to the folder and naming system.
Am I missing something?

Happy to see this project grow! :) Thanks for your work!

**Comment by Katzmann1983** [comment]
In the latest update, I changed the structure of the other_events.json, events_with_documents.json and account_transactions.csv
The transactions you are looking for can be generated using pytr:
`pytr export_transactions other_events.json account_transactions2.csv `

I am not using Portfolio Performance, so I am not to sure, how the input for it should look like and what needs to be a part of it.

**Reply by V0xxel** [reply]
My mistake. I did not realise that the visa payments have no documents provided by TR. So they will never appear in any folder 😅.

The provided command worked fine, thanks a lot. I've never seen this command before. Is there a reason for this "extra step"? 
I suspect that most casual users like myself mainly use the dl_docs command. This already produces the account_transactions.csv file, so I never suspected there was more to _unlock_ 😄 
Perhaps this could be merged or integrated into the dl_docs process?

Also, it would be nice to have the name of the vendor in the note column. Otherwise it just says "card transaction".

**Reply by V0xxel** [reply]
After further inspection of the two .csv files, I found the following 

- account_transactions.csv contains only the deposits made from a SEPA mandate (savings plans). _(It also contains the interest transactions that exist in duplicate as documents)_.
- account_transactions2.csv, on the other hand, only contains deposits that were manual bank transfers.

To complete the confusion, my last deposit, made after the TR update under their full bank licence (20.06.24), does not appear in either .csv file, but exists as a document in a "transaction confirmation" folder.

**Reply by Katzmann1983** [reply]
Interesting. I still don't have the new type of account. Can you share the json from the other_event.json corresponding to the missing transaction?
Currently I am writing a fix for the manual interaction and assuring that the import in portfolioperformance will be working fine.

**Reply by V0xxel** [reply]
The transaction is not included in the other_event.json but in the event_with_documents.json as it now comes with a proper PDF.
I've attached a json. 
[new_transfer_with_doc.json](https://github.com/user-attachments/files/16165471/new_transfer_with_doc.json)
The event type has changed from PAYMENT_INBOUND to INCOMING_TRANSFER.

Hope this helps :)
